django/db/models/sql/compiler.py

```diff
@@ -476,8 +476,9 @@ def as_nested_sql(self):
         Used when nesting this query inside another.
         """
         obj = self.query.clone()
-        if obj.low_mark == 0 and obj.high_mark is None and not self.query.distinct_fields:
-            # If there is no slicing in use, then we can safely drop all ordering
```

Comment:
I'd keep a comment above the if statement like: `# It's safe to drop ordering if the queryset isn't using slicing, distinct(*fields), or select_for_update().`